tests/src/com.github.ben-manes.caffeine/caffeine/2.9.3/src/test/java/com_github_ben_manes_caffeine/caffeine/RemovalTest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_github_ben_manes_caffeine.caffeine;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.RemovalCause;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class RemovalTest {
+    @Test
+    void testExplicitRemovals() {
+        Cache<String, String> cache = Caffeine.newBuilder().expireAfterWrite(10, TimeUnit.MINUTES).maximumSize(10_000).build();
+        cache.putAll(Map.of("Hello", "World", "Hi", "Universe", "Aloha", "Universe"));
+        assertThat(cache.getAll(List.of("Hello", "Hi", "Aloha"), key -> null))
+                .isEqualTo(Map.of("Hello", "World", "Hi", "Universe", "Aloha", "Universe"));
+        cache.invalidate("Hello");
+        assertThat(cache.getIfPresent("Hello")).isNull();
+        cache.invalidateAll(List.of("Hi", "Aloha"));
+        Stream.of("Hi", "Aloha").forEach(s -> assertThat(cache.getIfPresent(s)).isNull());
+        cache.invalidateAll();
```

Comment:
Isn't it that at this point all keys from the cache ("Hello", "Hi", "Aloha") are already invalidated? Therefore even without this line, [following assert](https://github.com/oracle/graalvm-reachability-metadata/pull/388/files#diff-50efd929831e90485065afdd094fbc8af2fb25f5987186c44c3c651b62c70f09R33) will always be correct? Maybe adding one more key-value pair to the Map (that won't be invalidated until this line) will test `cache.invalidateAll()` properly.

Reply:
- A new key-value pair is used. Solve.